service/tracker/execution.go

```diff
@@ -171,6 +189,45 @@ func (e *Execution) processNext() error {
 	return nil
 }
 
+func (e *Execution) checkTrieUpdates(record *uploader.BlockData, blockID flow.Identifier) error {
+	e.log.Debug().Hex("block_id", blockID[:]).Msg("fetching updates from data sync")
+	e.log.Debug().Int("updates", len(record.TrieUpdates)).Msg("got trie updates from GCP")
+	// get Trie updates from exec data sync
+	req := &access.GetExecutionDataByBlockIDRequest{BlockId: blockID[:]}
+	res, err := e.execClient.GetExecutionDataByBlockID(context.Background(), req)
+	if err != nil {
+		return fmt.Errorf("could not get execution data from access node: %w", err)
+	}
+
+	execTrieUpdates := make(map[string]bool, 0)
+	// collect updates before pushing!
+	for _, chunk := range res.GetBlockExecutionData().ChunkExecutionData {
+		convertedChunk, err := convert.MessageToChunkExecutionData(chunk, e.chain)
+		if err != nil {
+			return fmt.Errorf("unable to convert execution data chunk : %w", err)
+		}
+		if convertedChunk.TrieUpdate != nil {
+			execTrieUpdates[convertedChunk.TrieUpdate.String()] = true
+			e.log.Debug().Str("source", "exec sync").Str("update", convertedChunk.TrieUpdate.String()).Msg("")
+		}
+	}
+	// hash search for matching update
+	for _, gcpUpdate := range record.TrieUpdates {
+		if gcpUpdate != nil && !gcpUpdate.IsEmpty() {
```

Comment:
what if it is nil or not present? shouldn't we iterate over data fetched from execution data API and compare that if found in GCP data since we treat that as the source of truth. So instead of making a map out of API data, we make a map out of GCP data, then we get the data from API and iterate over all the data checking that every and each of the items is also found in GCP.